### DISCUSSION TITLE
Connect files using file list

### DISCUSSION THREAD

**Discussion by nktflppv** [ImageMagick/ImageMagick#6170] [Help]
### ImageMagick version

Version: ImageMagick 7.1.1-3 Q16-HDRI x86_64 20959 https://imagemagick.org

### Operating system, version and so on

mac os

### Description

Hi! I need for my project to connect 100,000 images in pairs. 1 image with 2, 3 merge 4, etc. I installed imagemagick and learned how to merge images manually using terminal macos
`convert +append 1.jpeg 2.jpeg +resize x500 newimage_1_2.jpeg`

I found the code, but from the description I understood that it does not fit, as it keeps the images in RAM and only connects them after going through the loop.

```
arr=(`ls *.jpeg`)
num=${#arr[*]}                                                             
for ((i=0; i<num; i=i+2)); do     
j=$((j+1))
k=$((k+2))
magick ${arr[$i]} ${arr[$j]} +append newimage_${j}_${k}.jpeg
done
```

But I don't understand how to connect the record file e.g.
```
1.jpeg 2.jpeg newimage_1_2.jpeg
3.jpeg 4.jpeg newimage_3_4.jpeg
5.jpeg 6.jpeg newimage_5_6.jpeg
7.jpeg 8.jpeg newimage_7_8.jpeg
9.jpeg 10.jpeg newimage_9_10.jpeg
11.jpeg 12.jpeg newimage_11_12.jpeg
13.jpeg 14.jpeg newimage_13_14.jpeg
15.jpeg 16.jpeg newimage_15_16.jpeg
17.jpeg 18.jpeg newimage_17_18.jpeg
19.jpeg 20.jpeg newimage_19_20.jpeg
21.jpeg 22.jpeg newimage_21_22.jpeg
23.jpeg 24.jpeg newimage_23_24.jpeg
25.jpeg 26.jpeg newimage_25_26.jpeg
27.jpeg 28.jpeg newimage_27_28.jpeg
29.jpeg 30.jpeg newimage_29_30.jpeg
31.jpeg 32.jpeg newimage_31_32.jpeg
```
Sorry, I'm not a programmer and don't understand where else to ask the question.  Thanks in advance for your help.

Translated with www.DeepL.com/Translator (free version)

**Comment by fmw42** [comment]
Proper IM syntax is to read the images before the command. So this works for me. I have images in desktop/test2 and want the output to be in desktop/test3.  Note that on IM 7, use magick not convert.

```
cd 
cd desktop/test2
arr=(`ls *.jpg`)
num=${#arr[*]}
for ((i=0; i<num-1; i=i+2)); do
j=$((i+1))
magick ${arr[$i]} ${arr[$j]} +append ../test3/newimage_${i}_${j}.jpg
done

```

Note that (`ls *.jpg`) only works if your images do not have spaces in the names.
